core/arch/arch_exports.h

```diff
@@ -157,14 +157,12 @@ typedef struct _spill_state_t {
     reg_t reg_stolen; /* slot for the stolen register */
 #elif defined(RISCV64)
     reg_t a0, a1, a2, a3;
-    /* Slot for the "stolen" register.
+    /* Slot for the stolen register, which is tp.
      * Note that on RISC-V, tp is the thread pointer, and it is also a general-purpose
-     * register, so unlike AARCHXX, we don't actually need to steal any register to store
-     * DR's tls, we can just use tp directly.
-     * But this also introduces another issue, we will lose the original value of
-     * tp. So in insert_load_dr_tls_base() we put the original value of tp into the x0
-     * slot of the dcontext, and in append_save_gpr() we restore the original value back
-     * to tp.
+     * register, so we steal tp to store DR's tls.
+     * But this also introduces another issue, we will lose the original value of tp. So
```

Comment:
This sentence can be removed; we're stealing the register which implies we're properly preserving the app's value.

Reply:
Removed.